tests/test_timefuncs.py

```diff
@@ -1042,23 +1042,11 @@ def timedelta64_and_arithmatic_operator() -> None:
     s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
     s3 = s2 - s1
     td = np.timedelta64(1, "M")
-    check(assert_type((s1 - td), TimestampSeries), np.timedelta64)
-    check(assert_type((s1 + td), pd.Series[Any]), np.timedelta64)
-    check(assert_type((s1 * td), TimedeltaSeries), np.timedelta64)
-    check(assert_type((s1 / td), pd.Series[float]), np.timedelta64)
-    check(
-        assert_type((s3 - td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 + td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 * td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 / td), pd.Series[float]),
-        pd.Series,
-    )
+    check(assert_type((s1 - td), TimestampSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s1 + td), TimestampSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s1 * td), TimedeltaSeries), pd.Series, pd.Timestamp)
```

Comment:
multipying a `TimestampSeries` times a `Timedelta` is not allowed, so have to fix that.

In `TimestampSeries`, put in a method for `__mul__()` that takes the two timedelta arguments, and returns `Never` from `typing_extensions`

Change the test to assert the type, but do not do the `check` part.

Reply:
after doing all this in mypy is having issues, it is saying `error: Expression is of type "NoReturn", not "TimestampSeries"  [assert-type]`